woocommerce/woocommerce-gateway-stripe
# For payments initiated by payment request buttons, set form top error container

# Changes proposed in this Pull Request:

Issue: Link to the GitHub issue this PR addresses (if appropriate).
Fixes #879 

Description: If a payment request button is used to initiate payment and fails (e.g. using a 3D Secure test card and then clicking FAIL when prompted) ensure the error is displayed to the user (and not just console logged.) To pull this off, we set a data flag on the payment request button itself when it is used to initiate payment. If an error occurs and that flag is present, it prompts us to use the woocommerce-notices-wrapper div as the error container.


Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.

<img width="1046" alt="Screen Shot 2020-12-07 at 3 53 08 PM" src="https://user-images.githubusercontent.com/1595739/101419641-4b5bd180-38a5-11eb-8fb2-d64acdde159b.png">
<img width="1027" alt="Screen Shot 2020-12-07 at 3 52 32 PM" src="https://user-images.githubusercontent.com/1595739/101419644-4d259500-38a5-11eb-8b3e-bd803f99bd69.png">
<img width="1001" alt="Screen Shot 2020-12-07 at 3 51 54 PM" src="https://user-images.githubusercontent.com/1595739/101419646-4dbe2b80-38a5-11eb-8796-6f7669c18843.png">


# Testing instructions

- Install this branch and activate it
- Ensure you are set up with keys and test mode for the Stripe plugin and that payment request buttons are enabled
- Add this filter:

`add_filter( 'wc_stripe_show_payment_request_on_checkout', '__return_true' );`

- Navigate to a product page using Chrome
- Click on the payment request button. When prompted, enter the 4000 0000 0000 3063 test card
- Enter a dummy CCV when prompted (e.g. 123)
- Wait for the Authorize/Fail modal. Click Fail.
- Ensure the error message is displayed
- Repeat for cart and checkout pages

-------------------

(remove me)
